src/wert-module.tsx

```diff
@@ -1,72 +1,116 @@
-import * as React from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
 import WertWidget from '@wert-io/widget-initializer';
+import type {
+  Options,
+  SetThemeParameters,
+  EventTypes,
+} from '@wert-io/widget-initializer/types';
 
-declare type WertPassThroughProps = React.ComponentProps<'div'>;
-declare type WertModuleProps = WertPassThroughProps & {
-  options: WertWidget["options"]
-};
-declare type WertModuleState = {
-  passThroughProps: WertPassThroughProps
-};
+import generateSignedData from './generate-signed-data';
+import type { SmartContractOptions } from './generate-signed-data';
+const reactiveThemeParameters = [
+  'theme',
+  'color_background',
+  'color_buttons',
+  'color_buttons_text',
+  'color_secondary_buttons',
+  'color_secondary_buttons_text',
+  'color_main_text',
+  'color_secondary_text',
+  'color_icons',
+  'color_links',
+  'color_success',
+  'color_warning',
+  'color_error',
+] as const;
+type ReactiveThemeParameters = typeof reactiveThemeParameters[number];
+export type StaticOptions = Omit<Options, ReactiveThemeParameters | 'listeners'>
+export type ReactiveOptions = Pick<Options, ReactiveThemeParameters | 'listeners'>
 
-class WertModule extends React.Component<WertModuleProps, WertModuleState> {
-  wertWidget: WertWidget;
-
-  constructor(props: WertModuleProps) {
-    super(props);
+export interface WidgetOptions {
```

Comment:
So, I've separated the static and reactive options, however, there are a couple of things I wanna clarify:

1. You've suggested passing the static options as a parameter to the open function. However, usually, I expect to pass the config to the hook and forget about it, and now the devs will have to keep it in mind. Also, if anyone used our initializer, they might've remembered that the open method doesn't expect anything. So, what do you think - may we keep it this way?
2. You've also suggested adding an isOpen boolean. However, it seems there's no way for us to understand when the widget is closed and removed from the DOM (apart from directly watching for an iframe). Or did you mean anything else by that?